### PR TITLE
feat: notification user avatar

### PR DESCRIPTION
**Pull request by setchy**
Took this idea from the GitHub Android application...

Display last updated user avatar.  Users name is available when mousing over.

Screenshot 2024-03-20 at 10 51 07 PM

### CODE REVIEW

**Comment by adufr** [comment]
This is a really cool addition!

I see that the PR is still in draft? Is there something missing to make it ready for review?

**Comment by setchy** [comment]
Still in draft. I need to overhaul the User-related types first.

**Comment by setchy** [comment]
Took a bit of git wrestling to get this branch back up-to-date...

appreciate feedback on the layout of the profile icon.  i played around with it being on the far left, but wasn't sure where the `Subscribed`, `Mentioned` notification reasons would be best displayed then...

File: .husky/pre-commit
Comment (bmulholland) [inline .husky/pre-commit:1]: What happened here?

File: src/components/NotificationRow.test.tsx [hunk 1]
```diff
@@ -19,8 +19,8 @@ describe('components/NotificationRow.tsx', () => {
     jest.clearAllMocks();
   });
 
-  it('should render itself & its children', async () => {
-    (global as any).Date.now = jest.fn(() => new Date('2014'));
+  it('should render itself & its children with avatar', async () => {
+    (global as any).Date.now = jest.fn(() => new Date('2024'));
```
Comment (bmulholland) [inline src/components/NotificationRow.test.tsx:23]: Poof! Modern.

**Review by bmulholland — APPROVED** [review]

File: .husky/pre-commit
Comment (adufr) [reply .husky/pre-commit:1]: This looks like Unix permissions changes 🤔

Comment (bmulholland) [reply .husky/pre-commit:1]: I think it's just a newline was removed at the end of the file

File: src/components/NotificationRow.tsx
[the remark this answers is not in the document]
Comment (bmulholland) [reply src/components/NotificationRow.tsx:1]: (Not a blocker! Just a thought on future direction.)

File: .husky/pre-commit
Comment (adufr) [reply .husky/pre-commit:1]: I don't think so, because of this:
image

`100755 → 100644` to me is related to permission changes

Comment (bmulholland) [reply .husky/pre-commit:1]: Then it's two things :)
Screenshot 2024-04-03 at 13 40 55
Hover state: "No newline at end of file"

File: src/components/NotificationRow.tsx
[the remark this answers is not in the document]
Comment (setchy) [reply src/components/NotificationRow.tsx:1]: 100% Agree - I'm still trying to wrap my head around how we can do this in bitsized chunks.  A worthy goal, for sure

**Comment by setchy** [comment]
How do we feel about the structure of that footer.  

It will follow two patterns, depending on whether the notification event contains user data.

1. `<reason>` - `<avatar img>` updated about `<some time>`
2. `<reason>` - Updated about `<some time`>

Some open questions that are going through my mind

- should the avatar image be at the start of the notification.  if so, where do we put the reason?
- should we have a default avatar image for unknown user.  that would ensure each row has a consistent shape.  currently it can vary depending on the notification content (example below)

Screenshot 2024-04-03 at 11 30 02 AM

**Comment by adufr** [comment]
I like that there's no avatar when the reason isn't a user (like for the worflow run notification in your last screenshot); it helps me quickly identify different kind of notifications

I don't really have an opinion on the placement of the avatar, it doesn't look too bad like this...
Or maybe having it at the end (like, "updated about 4 hours ago by `<avatar>`)..? I don't know, it looks ok to me like this!

**Comment by bmulholland** [comment]
I kinda like the avatar at the start, makes it more like an english sentence, "actor acted at time"

**Comment by setchy** [comment]
> I kinda like the avatar at the start, makes it more like an english sentence, "actor acted at time"

I'll mock up an example of what this may look like.

**Comment by setchy** [comment]
Examples of what an alternate layout could look like.

Format is <avatar> <reason type> <update at>, where
- avatar is user avatar when known, else fallback to person icon
- reason type used instead of previous hardcoded `updated`.  Each reason.type will be updated to be correct tense (ie: `Mention` -> `Mentioned`, etc) 
- updated at, no change

Screenshot 2024-04-05 at 1 09 01 PM

Screenshot 2024-04-05 at 1 10 12 PM

**Comment by bmulholland** [comment]
I love those, clean and clear :)

**Comment by afonsojramos** [comment]
Looks awesome!! Let me run this locally to check it out better!

**Comment by setchy** [comment]
> Looks awesome!! Let me run this locally to check it out better!

Thanks mate. Appreciate any feedback, particularly on the changes to the new reason mappings.

**Review by afonsojramos — APPROVED** [review]
We lose the usernames, but we get the avatars, I think this is a positive change overall, love it!

Clean implementation as well, all wins!

**Comment by setchy** [comment]
> We lose the usernames, but we get the avatars, I think this is a positive change overall, love it!
> 
> Clean implementation as well, all wins!

thanks @afonsojramos, appreciate the feedback.  usernames still available on avatar hover :)

**Comment by afonsojramos** [comment]
> > We lose the usernames, but we get the avatars, I think this is a positive change overall, love it!
> > 
> > Clean implementation as well, all wins!
> 
> thanks @afonsojramos, appreciate the feedback.  usernames still available on avatar hover :) 

I know, but we still lose them at a glance, which is mostly what matters.

**Pull request merged**